src/core/transaction.c

```diff
@@ -1102,6 +1102,12 @@ int transaction_add_job_and_dependencies(
         }
 
         return 0;
+fail:
+        /* Recursive call failed to add required jobs so let's drop top level job as well. */
+        log_unit_debug_errno(unit, r, "Cannot add dependency job: %s", bus_error_message(e, r));
```

Comment:
maybe reword to say what you are doing here as effect, i.e. something like `log_unit_debug_errno(…, "Unable to add dependency job to transaction, deleting job %s/%s again", unit->id job_type_to_string(type)));`

you get the idea